File: backend-python/app/schemas/job.py
"""JobForge AI - Job Schemas"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from uuid import UUID

class JobBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    company: str = Field(..., min_length=1, max_length=255)
    location: str = Field(..., min_length=1, max_length=255)
    remote_type: Optional[str] = None
    description: str
    requirements: Optional[str] = None
    raw_description: Optional[str] = None
    ai_summary: Optional[str] = None
    ai_highlights: Optional[List[str]] = None
    ai_required_skills: Optional[List[str]] = None
    ai_compensation: Optional[str] = None
    ai_remote_policy: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    job_type: Optional[str] = None
    experience_level: Optional[str] = None
    source_url: Optional[str] = None
    validated_source_url: Optional[str] = None
    source_site: Optional[str] = None

class JobCreate(JobBase):
    posted_date: Optional[datetime] = None

class JobUpdate(BaseModel):
    title: Optional[str] = None
    company: Optional[str] = None
    location: Optional[str] = None
    remote_type: Optional[str] = None
    description: Optional[str] = None
    requirements: Optional[str] = None
    raw_description: Optional[str] = None
    ai_summary: Optional[str] = None
    ai_highlights: Optional[List[str]] = None
    ai_required_skills: Optional[List[str]] = None
    ai_compensation: Optional[str] = None
    ai_remote_policy: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    job_type: Optional[str] = None
    experience_level: Optional[str] = None
    source_url: Optional[str] = None
    validated_source_url: Optional[str] = None
    source_site: Optional[str] = None
    is_active: Optional[bool] = None

class JobResponse(JobBase):
    id: UUID
    is_active: bool
    posted_date: Optional[datetime] = None
    ai_last_enriched_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True

File: backend-go/internal/models/job.go
package models

import (
	"time"
)

// Job represents a job posting
type Job struct {
	ID                 string     `json:"id"`
	Title              string     `json:"title"`
	Company            string     `json:"company"`
	Location           string     `json:"location"`
	RemoteType         *string    `json:"remote_type"`
	Description        string     `json:"description"`
	Requirements       *string    `json:"requirements"`
	RawDescription     *string    `json:"raw_description"`
	SalaryMin          *float64   `json:"salary_min"`
	SalaryMax          *float64   `json:"salary_max"`
	JobType            *string    `json:"job_type"`
	ExperienceLevel    *string    `json:"experience_level"`
	SourceURL          *string    `json:"source_url"`
	ValidatedSourceURL *string    `json:"validated_source_url"`
	SourceSite         *string    `json:"source_site"`
	AISummary          *string    `json:"ai_summary"`
	AIHighlights       []string   `json:"ai_highlights,omitempty"`
	AIRequiredSkills   []string   `json:"ai_required_skills,omitempty"`
	AICompensation     *string    `json:"ai_compensation"`
	AIRemotePolicy     *string    `json:"ai_remote_policy"`
	IsActive           bool       `json:"is_active"`
	PostedDate         *time.Time `json:"posted_date"`
	AILastEnrichedAt   *time.Time `json:"ai_last_enriched_at"`
	CreatedAt          time.Time  `json:"created_at"`
}

// JobSearchParams represents search parameters
type JobSearchParams struct {
	Query     string  `form:"query"`
	Location  string  `form:"location"`
	Remote    bool    `form:"remote"`
	JobType   string  `form:"job_type"`
	MinSalary float64 `form:"min_salary"`
	MaxSalary float64 `form:"max_salary"`
	Page      int     `form:"page"`
	Limit     int     `form:"limit"`
}

// ScraperStatus represents scraper health
type ScraperStatus struct {
	IsRunning       bool      `json:"is_running"`
	LastRun         time.Time `json:"last_run"`
	JobsScraped     int       `json:"jobs_scraped"`
	SuccessfulSites []string  `json:"successful_sites"`
	FailedSites     []string  `json:"failed_sites"`
}
